src/settings.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2011-2018 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <cstdint>
+#include <bitcoin/bitcoin/settings.hpp>
+
+namespace libbitcoin {
+
+// Common default values (no settings context).
+settings::settings()
+  : retargeting_factor(4),
+    target_spacing_seconds(10 * 60),
+    easy_spacing_seconds(20 * 60),
+    timestamp_future_seconds(2 * 60 * 60),
+    target_timespan_seconds(2 * 7 * 24 * 60 * 60),
+    retarget_proof_of_work_limit(0x1d00ffff),
+    no_retarget_proof_of_work_limit(0x207fffff),
+    min_timespan(target_timespan_seconds / retargeting_factor),
+    max_timespan(target_timespan_seconds * retargeting_factor),
+    retargeting_interval(target_timespan_seconds / target_spacing_seconds)
+{
+}
+
+uint32_t settings::work_limit(bool retarget) const
```

Comment:
We should move this function out of settings and into chain::header (see chain::header::proof). Prob should do that in an independent commit.

Reply:
Independent commit as in independent PR?

Reply:
@evoskuil I've been now looking into moving this function into chain::header as you've suggested. The problem is that it is also used in [chain::chain_state](https://github.com/libbitcoin/libbitcoin/blob/2e1e6792736e73f8a216ea09038084c01f22972a/src/chain/chain_state.cpp#L604) where there's no access to a `header` object. However, there's a comment which states that `bits.self` is unused. If this assignment can be removed then moving `work_limit()` becomes trivial.

Reply:
Yes, I you can set it to zero in that case. Self has no purpose in the context of a pool chain state and when the pool state is upgraded to a block the blocks bits are populated.